sqlx-sqlite/src/connection/explain.rs

```diff
@@ -428,13 +428,21 @@ pub(super) fn explain(
 
     let mut visited_branch_state: HashSet<BranchStateHash> = HashSet::new();
 
+    let mut gas = 500_000;
     let mut result_states = Vec::new();
 
     while let Some(mut state) = states.pop() {
         while state.program_i < program_size {
             let (_, ref opcode, p1, p2, p3, ref p4) = program[state.program_i];
             state.history.push(state.program_i);
 
+            //limit the number of 'instructions' that can be evaluated
+            if gas > 0 {
+                gas -= 1;
+            } else if gas == 0 {
```

Comment:
Why not just `else { break; }`?

Every time I see something like this I get paranoid that something is going to make it leapfrog 0 and just keep going.

Reply:
That's a hold over of an unimplemented plan to make gas configurable through an environment variable, where negative initial gas served to disable the limit entirely.  But it isn't configurable, so swapping for bare else is fine.